### ISSUE TITLE
False positive on `UnusedImport` with certain kotlinx-serialization 1.9.0 methods

### ISSUE THREAD

**Issue by gobetti** [detekt/detekt#8443]
## Expected Behavior
No unused imports reported.

## Observed Behavior
After updating kotlinx-serialization from 1.8.1 to 1.9.0, these imports are incorrectly flagged as unused everywhere they're used:
```
import kotlinx.serialization.json.add
import kotlinx.serialization.json.decodeFromJsonElement
import kotlinx.serialization.json.encodeToJsonElement
```

However, removing those imports causes the build to fail, i.e. they're actually used.

## Steps to Reproduce

Code example using `add`
```
import kotlinx.serialization.json.add
import kotlinx.serialization.json.buildJsonObject

buildJsonObject {
  putJsonArray("languages") {
    add(LocaleManager.languageTag)
  }
}
```

Code example using `decodeFromJsonElement` and `encodeToJsonElement`:
```
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.decodeFromJsonElement
import kotlinx.serialization.json.encodeToJsonElement

@Serializable data class Foo(val key: String)

val object = buildJsonObject {
  put("key", "Hello")
}
val foo: Foo = Json.decodeFromJsonElement(object)
val encoded = Json.encodeToJsonElement(foo)
```

## Context
Not an issue until `org.jetbrains.kotlinx:kotlinx-serialization-json:1.9.0`. Downgrading to 1.8.1 silences those warnings.

## Your Environment
* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 9.0.0
* Gradle scan link (add `--scan` option when running the gradle task): NA
* Operating System and version: I can repro on both macOS and Ubuntu
* Link to your project (if it's a public repository): NA

**Comment by atulgpt** [comment]
Hi @gobetti kotlinx serialization lib doesn't have any changes in those versions around the methods you mentioned. Are you sure other dependencies, like Kotlin version, detekt version are same across two project(one where issue is happening and one where it is not)? \
Also can you create a minimal reproducer of the issue?

**Comment by gobetti** [comment]
hey @atulgpt , thank you for the answer. Yes, I'm sure that the only change between having the issue, and not having the issue in my project, is the 1 line in my libs.versions.toml file, where `org.jetbrains.kotlinx:kotlinx-serialization-json:1.8.1` is changed to `org.jetbrains.kotlinx:kotlinx-serialization-json:1.9.0`

however, adding that dependency to the detekt project and using my provided examples does not trigger any errors with `./gradlew detektMain`. I did confirm that this task does report unused imports if I force some, but it doesn't report any unused imports with my examples, which is the correct behavior; unfortunately, it behaves differently in my project

after further investigation, I observed that the detekt project is using kotlin 2.2.10, while my project is currently on 2.1.21; if I update to 2.2.10, then I don't see the `UnusedImports` warnings I'm reporting here, but it makes me get the ones from #8212 (`UnnecessaryAnnotationUseSiteTarget`)

if I deactivate `UnnecessaryAnnotationUseSiteTarget` in my detekt.yml config, then no issues are reported.

**Conclusion:** the false positives seen with kotlinx-serialization 1.9.0 should get fixed by updating to Kotlin 2.2.10
